iroh-net/src/discovery/local_node_discovery.rs

```diff
@@ -0,0 +1,250 @@
+                    Err(err) => {
+                        tracing::error!("LocalNodeDiscovery service error: {err:?}");
+                        tracing::error!("closing LocalNodeDiscovery");
+                        timeouts.abort_all();
+                        return;
+                    }
+                    Ok(msg) => msg,
+                };
+                match msg {
+                    Message::Discovery((discovered_node_id, peer_info)) => {
+                        tracing::trace!(
+                            ?discovered_node_id,
+                            ?peer_info,
+                            "LocalNodeDiscovery Message::Discovery"
+                        );
+                        let discovered_node_id = match PublicKey::from_str(&discovered_node_id) {
+                            Ok(node_id) => node_id,
+                            Err(e) => {
+                                tracing::warn!(
+                                    discovered_node_id,
+                                    "couldn't parse node_id from mdns discovery service: {e:?}"
+                                );
+                                continue;
+                            }
+                        };
+
+                        if discovered_node_id == node_id {
+                            continue;
+                        }
+
+                        if peer_info.is_expiry() {
+                            tracing::trace!(
+                                ?discovered_node_id,
+                                "removing node from LocalNodeDiscovery address book"
+                            );
+                            node_addrs.remove(&discovered_node_id);
+                            continue;
+                        }
+
+                        if let Some(sender) = senders.get(&discovered_node_id) {
+                            let item: DiscoveryItem = (&peer_info).into();
+                            tracing::trace!(?item, "sending DiscoveryItem");
+                            sender.send_async(Ok(item)).await.ok();
+                        }
+                        tracing::trace!(
+                            ?discovered_node_id,
+                            ?peer_info,
+                            "adding node to LocalNodeDiscovery address book"
+                        );
+                        node_addrs.insert(discovered_node_id, peer_info);
+                    }
+                    Message::SendAddrs((node_id, sender)) => {
+                        tracing::trace!(?node_id, "LocalNodeDiscovery Message::SendAddrs");
+                        if let Some(peer_info) = node_addrs.get(&node_id) {
+                            let item: DiscoveryItem = peer_info.into();
+                            tracing::debug!(?item, "sending DiscoveryItem");
+                            sender.send_async(Ok(item)).await.ok();
+                        }
+                        senders.insert(node_id, sender);
```

Comment:
should this only happen in case no address is known yet?

Reply:
I left this here in case the discovery service wanted to listen for the full 10s before it times out, in case of address changes. Sometimes discovery just pulls the first address it gets off the stream, but it can also add any subsequent addresses it gets to the underlying node map that we have in our endpoint, if it doesn't cut off the stream early.